### DISCUSSION TITLE
Blazor WebAssembly mode and PWA

### DISCUSSION THREAD

**Discussion by pnicosia80** [oqtane/oqtane.framework#3173] [Q&A]
Good morning,
I would like to configure a module to work client side with webassemblies. In the site settings I have already set the Blazor WebAssembly mode and the PWA, what should I set in the module to make it work on the client side? I see that the broswer doesn't download anything, even without PWA.
Also DataGrid syncfusion renders data only in server mode, where am I wrong?
I also take the opportunity to ask for clarification on PWA functionality.
Finally how does the Blazor Hybrid work?

Thanks in advance

**Comment by sbwalker** [comment]
@adrianus80 I am not sure you understand the different Blazor hosting models...

- Blazor WebAssembly is a client-side hosting model which downloads .NET assemblies to the browser and runs them - it has no relation to a PWA
- a PWA relies on a Service Worker (JavaScript) and can be hosted either on Blazor WebAssembly or Blazor Server
- Blazor Hybrid allows you to run Razor components in a desktop or mobile application using a WebView
- if you are building Oqtane modules which need to run on WebAssembly then you need to define the Dependencies in the ModuleInfo,cs file - as that is how Oqtane is able to determine which assemblies need to be downloaded to the browser 
- if you are using Syncfusion components on WebAssembly then you will need add Syncfusion to the Dependencies property
- when running on WebAssembly, Oqtane downloads all assemblies using an API call which bundles the DLLs into a single ZIP file - so you will not see individual assemblies being downloaded in your browser networking tools
